consensus/src/extension/election.rs

```diff
@@ -0,0 +1,333 @@
+use std::collections::HashMap;
+
+use crate::{
+    extension::{units::Units, ExtenderUnit},
+    Hasher, NodeCount, NodeIndex, NodeMap, Round,
+};
+
+fn common_vote(relative_round: Round) -> bool {
+    // This should only be called for relative round >= 2, so to be precise we start with true, false, true, and then
+    if relative_round == 3 {
+        return false;
+    }
+    if relative_round <= 4 {
+        return true;
+    }
+    // we alternate between true and false starting from true in round 5.
+    relative_round % 2 == 1
+}
+
+enum VoteError<H: Hasher> {
+    EliminateCandidate,
+    ElectionDone(H::Hash),
```

Comment:
Yeah, this isn't an error and that also annoys me somewhat. If I made the result of voting into an `enum` there would be a lot of `match`es in methods that completely don't need them because `Carrier` is [not yet stabilized](https://doc.rust-lang.org/1.17.0/std/ops/trait.Carrier.html); plus I would have to parametrize it, since the `Ok` return types differ. ;_; Maybe just renaming this type would help? To, uh, `CandidateOutcome` maybe?

The non-reusability is definitely by design. I don't think the performance concerns here matter much in practice – we almost always elect the "first" candidate anyway.